src/main/java/com/opentable/versionedconfig/GitOperations.java

```diff
@@ -205,4 +218,26 @@ private String relevantDiffPath(DiffEntry diff) {
             throw new VersioningServiceException("Can't get diff", e);
         }
     }
+
+    private <T> T upstreamRetry(Function<Integer, T> action) {
```

Comment:
I think retry is a bit confusing, since to me retry means trying the same thing again. But, I don't really know of a better name, maybe `tryExecuteOnRemotes`?

Reply:
Well, it is trying the same thing again... against a different origin server.  It'll only do a second try if the first one fails with an exception.  I think that fits within the definition of `retry` but I'll think more about the name.